libs/design/tree/src/tree/tree.component.ts

```diff
@@ -154,4 +170,18 @@ export class DaffTreeComponent extends _daffTreeBase implements OnInit, OnChange
       this.flatTree = flattenTree(this._tree, this.renderMode === 'not-in-dom');
     });
   }
+
+  ngAfterViewInit(): void {
+    this.items.changes.pipe(
+      takeUntilDestroyed(this.destroyRef),
+    ).subscribe((items: QueryList<DaffTreeItemDirective>) => {
+      runInInjectionContext(this.injector, () => afterNextRender(() => {
+        const activeTreeItem = items.find((treeItem) => treeItem.selected);
+        if (activeTreeItem) {
+          activeTreeItem.openAncestors();
+          this.cd.markForCheck();
```

Comment:
I don't believe that a `markForCheck` is sufficient here. It may be that we have to do a full `detectChanges`. I would expect Angular to error here with a `Expression changed after is has been checked`.

Reply:
I haven't noticed that error in the console. I can still change it if you want though

Reply:
I think the task queue shift created by the observable is likely hiding the error.